src/modules/consul.c

```diff
@@ -562,20 +716,24 @@ static void process_global_curlm(void) {
         mtevL(error_ls, "%s -> %s\n", done_url, curl_easy_strerror(message->data.result));
       }
       long httpcode;
-      curl_easy_getinfo(easy_handle, CURLINFO_PRIVATE, (char **)&dyn);
+      curl_easy_getinfo(easy_handle, CURLINFO_PRIVATE, (char **)&handler);
+      if(handler->handler) {
+        handler->handler(message->data.result, easy_handle, &handler->dyn, handler->userdata);
+      }
       curl_easy_getinfo(easy_handle, CURLINFO_RESPONSE_CODE, &httpcode);
       if(httpcode == 200) {
         mtevL(debug_ls, "%s DONE\n", done_url);
       } else {
-        mtevL(error_ls, "%s -> %d %.*s\n", done_url, (int)httpcode,
-              (int)mtev_dyn_buffer_used(dyn), (const char *)mtev_dyn_buffer_data(dyn));
+        mtevL(debug_ls, "%s -> %d %.*s\n", done_url, (int)httpcode,
+              (int)mtev_dyn_buffer_used(&handler->dyn),
+              (const char *)mtev_dyn_buffer_data(&handler->dyn));
       }
  
       curl_multi_remove_handle(global_curl_handle, easy_handle);
       curl_easy_cleanup(easy_handle);
-      if(dyn) {
-        mtev_dyn_buffer_destroy(dyn);
-        free(dyn);
+      if(handler) {
+        mtev_dyn_buffer_destroy(&handler->dyn);
```

Comment:
previously we would free "dyn" after mtev_dyn_buffer_destroy, but we no longer do.  Was this no longer necessary?

Reply:
dyn is no longer allocated, it's now an incorporated member of the handler struct.